crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -256,6 +259,11 @@ pub trait BuildChildren {
     fn remove_children(&mut self, children: &[Entity]) -> &mut Self;
     /// Adds a single child
     fn add_child(&mut self, child: Entity) -> &mut Self;
+
+    /// Set the parent.
+    ///
+    /// This overwrites the existing parent.
```

Comment:
...I assume, for all children of the underlying collection at once?

Reply:
The idea is that

```rust
child.set_parent(parent)
// is equivalent to
parent.add_child(child)
```

I don't understand what you mean by "for all children of the underlying collection". Could you precise?

If you mean the `Children` component of the `parent`, the answer is no. `set_parent` just sets the parent of an entity, so it does nothing else than:

1. Add a `Parent(parent)` component to the `child`
2. Push `child` to the `Children` component of the `parent`